src/commands/generate/types.ts

```diff
@@ -0,0 +1,86 @@
+import { CSharpFileGenerator, JavaFileGenerator, JavaScriptFileGenerator, TypeScriptFileGenerator, GoFileGenerator, Logger} from '@asyncapi/modelina';
+import { Flags } from '@oclif/core';
+import Command from '../../base';
+import { load } from '../../models/SpecificationFile';
+import { parse } from '@asyncapi/parser';
+export default class Types extends Command {
+  static description = 'Generates typed models';
+
+  static args = [
+    { name: 'language', description: 'language output', options: ['typescript', 'csharp', 'golang', 'java', 'javascript'], required: true},
+    { name: 'file', description: 'spec path, url, or context-name', required: true },
+  ]
+
+  static flags = {
+    help: Flags.help({ char: 'h' }),
+    output: Flags.string({ char: 'o', description: 'output path to write the types to', required: true}),
```

Comment:
should not be required, if `--output` is not provided, then print result to terminal. In AsyncAPI CLI flags always are optional

Reply:
But types generation in console (and also templates) can be a really big problem. I understand that flags are optional, but in this case output is very important. Additionally I don't like this solution in cli from generator where for output we have fallback in form of generating in given folder where we run cli - it's not a good UX solution for me.

Reply:
what is your suggestion though? output as argument?

Reply:
I agree that it shouldn't be required and default to printing on `stdout` (console). This way people can use the CLI to pipe the output to another tool:

```bash
asyncapi generate types | sed ... | etc.
```

Reply:
Yes, but generating types and templates is usually a dozen or so files, how do you want to handle it on stdout? Is it even possible to iterate by array?

Reply:
Can't you print all of them? Just like `git diff` does:

```
### File: MyClass.java

class MyClass {
...
}

### File: MyOtherClass.java

class MyOtherClass {
...
}
```

Reply:
Just what value will it have for the user himself? I can understand if someone would operate on a single file and not on a collection of files. I'm afraid it will only cause problems.

Reply:
Another solution would be to generate (without `output` flag) things in a folder named `generated-{timestamp}` and in the output we would give the name of this folder. Of course that folder will be located in current working directory.

Reply:
> I already explained the value. People can chain commands to manipulate the output. Useful for CI environments for instance, where you may want to replace some values from the output. But I understand your point. I'd not worry about these details in this PR but something to consider long-term. A CLI that can't be chained in the command line is a pain in the ass.

I agree, that we need to support chaining. 

I did not realize that it should be possible to output for stdout 🤔 I dont think that it's useful without it being documented how it is possible to chain it, or at least give some examples, otherwise we risk that it won't be possible at all. For example for the models, how do you split up the models and access the information needed to further your chaining? I.e. accessing the model names and the content of the model and iterating over them in shell scripts? 🤔